### PR TITLE
Refactored mujoco reach_env.

### PR DESCRIPTION
**Pull request by utkarshjp7**

### CODE REVIEW

File: sawyer/mujoco/robots/sawyer.py [hunk 1]
```diff
@@ -117,13 +117,11 @@ def gripper_position(self):
...
         state = state * 2 - 1
         return state
 
     @gripper_state.setter
     def gripper_state(self, state):
         state = (state + 1.) / 2.
-        self._env.sim.data.ctrl[:2] = np.array([
-            state * self.GRIPPER_STATE_SCALE,
-            -state * self.GRIPPER_STATE_SCALE])
+        self._env.sim.data.ctrl[0] = state * self.GRIPPER_STATE_SCALE
```
Comment (akumaraguru) [inline sawyer/mujoco/robots/sawyer.py]: Does this actuate both gripper fingers? It looks like you're only editing one of the finger positions here.

File: sawyer/mujoco/envs/reacher_env.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+
+from gym.spaces.box import Box
+
+from sawyer.garage.core import Serializable
+from sawyer.garage.envs.mujoco import MujocoEnv
+from sawyer.garage.envs.mujoco.mujoco_env import MODEL_DIR
+from sawyer.garage.misc.overrides import overrides
+from sawyer.mujoco.robots import PositionSpaceSawyer, TaskSpaceSawyer
+
+REACHERENV_COLLISION_WHITELIST = [
```
Comment (akumaraguru) [inline sawyer/mujoco/envs/reacher_env.py]: Make sure this whitelist adds to COLLISION_WHITELIST (see ToyEnv). An empty whitelist will lead to lots of spurious collisions between the sawyer's joints.

File: sawyer/mujoco/envs/reacher_env.py [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+
+    @property
+    @overrides
+    def action_space(self):
+        return self._robot.action_space
+
+    @property
+    @overrides
+    def observation_space(self):
+        spaces = []
```
Comment (akumaraguru) [inline sawyer/mujoco/envs/reacher_env.py]: You can just return `self._robot.observation_space` here.

File: sawyer/mujoco/envs/reacher_env.py [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+        elif self._xml_config == 'task':
+            obs.append(robot_obs['sawyer_gripper_position'])
+            obs.append(robot_obs['sawyer_gripper_state'])
+            obs = np.concatenate(obs).ravel()
+        else:
+            warnings.warn("Don't know how to pull observation data for config:"
+                " {}".format(self._xml_config))
+        return obs
+
+    def compute_reward(self, info):
```
Comment (akumaraguru) [inline sawyer/mujoco/envs/reacher_env.py]: Add support for a sparse reward mode as well.

File: sawyer/mujoco/robots/sawyer.py (hunk 1 above)
Comment (utkarshjp7) [reply sawyer/mujoco/robots/sawyer.py]: You are right, it doesn't. I undid my changes.

File: sawyer/mujoco/envs/reacher_env.py (hunk 2 above)
Comment (utkarshjp7) [reply sawyer/mujoco/envs/reacher_env.py]: Done.

File: sawyer/mujoco/envs/reacher_env.py (hunk 3 above)
Comment (utkarshjp7) [reply sawyer/mujoco/envs/reacher_env.py]: Done.

File: sawyer/mujoco/envs/reacher_env.py (hunk 4 above)
Comment (utkarshjp7) [reply sawyer/mujoco/envs/reacher_env.py]: Done.

**Comment by utkarshjp7** [comment]
Tested with following launcher
```
from garage.baselines import LinearFeatureBaseline
from garage.envs import normalize
from garage.experiment import LocalRunner
from garage.tf.algos import TRPO
from garage.tf.policies import GaussianMLPPolicy
from garage.tf.envs import TfEnv
from sawyer.mujoco.envs import ReacherEnv

with LocalRunner() as runner:
    env = TfEnv(normalize(ReacherEnv(goal_position=[0.3, 0.3, 0.3], xml_path='reacherenv_task.xml', xml_config='task')))

    policy = GaussianMLPPolicy(
        name="policy", env_spec=env.spec, hidden_sizes=(32, 32))

    baseline = LinearFeatureBaseline(env_spec=env.spec)
    algo = TRPO(
        env=env,
        policy=policy,
        baseline=baseline,
        max_path_length=100,
        discount=0.99,
        max_kl_step=0.01,
    )
    runner.setup(algo, env)
    runner.train(n_epochs=100, batch_size=4000, plot=True)
```

**Pull request merged**